grind.sh

```diff
@@ -159,10 +159,13 @@ upload_deployment_events() {
 
   temp_json="$(mktemp "/tmp/deploy_events_XXXXXX").json"
 
-  if jq -e type | grep -q array; then
-    cat > "$temp_json"
+  #pull the passed in info into a variable to allow multi-step processing
+  temp_cache=$(cat)
```

Comment:
This will need quotes to not leak if the input contains spaces

```suggestion
  temp_cache="$(cat)"
```

Reply:
However, reading the entire input into a variable doesn't seem like a good idea — there's a temp file right there. We should just use that. Then `jq` can be run on it as many times and ways as needed.

That said, we could just have the user tell us which format it's in and read it appropriately and stop doing magic. If we're server-side, maybe that's valuable, but the user's _right there_, and can tell us things.

Reply:
Fair points. Let me rework

Reply:
or we could just support a single format, and let users mush it into that format